geode-pulse/src/main/java/org/apache/geode/tools/pulse/internal/security/OauthSecurityConfig.java

```diff
@@ -0,0 +1,100 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.tools.pulse.internal.security;
+
+import static org.springframework.security.config.Customizer.withDefaults;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.oauth2.client.InMemoryOAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.AuthenticatedPrincipalOAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+@Profile("pulse.authentication.oauth")
+@PropertySource("classpath:pulse.properties")
+public class OauthSecurityConfig extends WebSecurityConfigurerAdapter {
+  @Value("${pulse.oauth.provider}")
+  private String providerId;
+  @Value("${pulse.oauth.clientId}")
+  private String clientId;
+  @Value("${pulse.oauth.clientSecret}")
+  private String clientSecret;
+  @Value("${pulse.oauth.authorizationUri}")
+  private String authorizationUri;
+  @Value("${pulse.oauth.tokenUri}")
+  private String tokenUri;
+  @Value("${pulse.oauth.userInfoUri}")
+  private String userInfoUri;
+  @Value("${pulse.oauth.jwkSetUri}")
+  private String jwkSetUri;
+  @Value("${pulse.oauth.userNameAttributeName}")
+  private String userNameAttributeName;
+
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    http.authorizeRequests(authorize -> authorize
+        .anyRequest().authenticated())
+        .oauth2Login(withDefaults());
+  }
+
+  @Bean
+  public ClientRegistrationRepository clientRegistrationRepository() {
+    return new InMemoryClientRegistrationRepository(this.uaaClientRegistration());
```

Comment:
It looks like we only have uaa support here. So if a customer is using oauth2 will it only work with uaa or can the configure it for some other oauth2 authentication server?

Reply:
actually that name can be changed